Text/URI/Types.hs

```diff
@@ -117,6 +124,70 @@ makeAbsolute scheme URI {..} = URI
 isPathAbsolute :: URI -> Bool
 isPathAbsolute = either id (const True) . uriAuthority
 
+-- | 'relativeTo ref base' makes a new 'URI' from 'ref' relative to absolute URI 'base'.
+--
+-- See also: <https://tools.ietf.org/html/rfc3986#section-5.2>
+
+relativeTo :: URI -> URI -> URI
+relativeTo r base
+    | isJust (uriScheme r) =
+        r { uriPath      = removeDotSegments (uriPath r) }
+    | isRight (uriAuthority r) =
+        r { uriScheme    = uriScheme base
+          , uriPath      = removeDotSegments (uriPath r)
+          }
+    | isNothing (uriPath r) =
+        r { uriScheme    = uriScheme base
+          , uriAuthority = uriAuthority base
+          , uriPath      = uriPath base
+          , uriQuery     = if not (null (uriQuery r))
+                             then uriQuery r
+                             else uriQuery base
+          }
+    | isPathAbsolute r =
+        r { uriScheme    = uriScheme base
+          , uriAuthority = uriAuthority base
+          , uriPath      = removeDotSegments (uriPath r)
+          }
+    | otherwise =
+        r { uriScheme    = uriScheme base
+          , uriAuthority = uriAuthority base
+          , uriPath      = removeDotSegments (mergePaths base r)
+          }
+
+mergePaths :: URI -> URI -> Maybe (Bool, NonEmpty (RText 'PathPiece))
+mergePaths base' r'
+  | isRight (uriAuthority base') && isNothing (uriPath base') = uriPath r'
```

Comment:
> Not sure why r' and base' are used when we could use r and base. 

It's because GHC (or Hlint, don't remember) complains when you reuse names of existing bindings that are in scope.

Reply:
But you don't have `base` or `r` in scope here. Maybe you separated this function later though...

Reply:
Yeah, you're right, I've seperated the function later. So you were right that we can use `r` and `base`.